### PR TITLE
allow additional certificates to be added to a pkcs7

### LINKED ISSUE #5492: Add additional x509 certificate to SMIMESignatureBuilder
As mentioned in #5433: It would be great to be able to add an additional certificate when using the new SMIMESignatureBuilder API. 

The background is that my x509 certificate is from a commercial CA. But - as it's commonly the case - it is not signed directly by the Root CA (instead it's signed from an intermediate/sub CA). So when I send a message to a new contact I would definitely want to include my certificate AND the certificate of the intermediate CA.

### PR DESCRIPTION
**Pull request by reaperhulk**
fixes #5492

### CODE REVIEW

File: docs/hazmat/primitives/asymmetric/serialization.rst [hunk 1]
```diff
@@ -647,6 +647,13 @@ contain certificates, CRLs, and much more. PKCS7 files commonly have a ``p7b``,
             :class:`~cryptography.hazmat.primitives.hashes.SHA384`, or
             :class:`~cryptography.hazmat.primitives.hashes.SHA512`.
 
+    .. method:: add_certificate(certificate)
+
+        Add an additional certificate to the PKCS7 structure. This method may
```
Comment (alex) [inline docs/hazmat/primitives/asymmetric/serialization.rst:652]: Docs ought to describe what the certs are for :-)

Also, does it just put all of these in a ball? They don't need to be assosciated with a particular signer?

Comment (reaperhulk) [reply docs/hazmat/primitives/asymmetric/serialization.rst:652]: If PKCS7 allows association it's not in the OpenSSL API. Big ball best ball. Let the verifier sort 'em out I always say.

**Review by alex — APPROVED** [review]

File: src/cryptography/hazmat/primitives/serialization/pkcs7.py [hunk 2]
```diff
@@ -63,6 +64,14 @@ def add_signer(self, certificate, private_key, hash_algorithm):
             self._signers + [(certificate, private_key, hash_algorithm)],
         )
 
+    def add_certificate(self, certificate):
+        if not isinstance(certificate, x509.Certificate):
+            raise TypeError("certificate must be a x509.Certificate")
```
Comment (alex) [inline src/cryptography/hazmat/primitives/serialization/pkcs7.py:69]: untested path

**Review by alex — CHANGES_REQUESTED** [review]
coverage

File: src/cryptography/hazmat/primitives/serialization/pkcs7.py (hunk 2 above)
Comment (reaperhulk) [reply src/cryptography/hazmat/primitives/serialization/pkcs7.py:69]: fixed

**Review by alex — APPROVED** [review]

**Pull request merged**